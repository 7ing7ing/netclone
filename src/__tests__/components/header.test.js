import React from "react";
import { fireEvent, render } from "@testing-library/react";
import { Header } from "../../components";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";

describe("<Header/>", () => {
  it("Renders the basic <Header/> with a background", () => {
    const history = createMemoryHistory();
    const { container, getByText, getByTestId } = render(
      <Router location={history.location} navigator={history}>
        <Header>
          <Header.Frame>
            <Header.Logo to="/" usrc="/logo.svg" alt="Netflix" />
            <Header.TextLink active="true">Hello I am a link!</Header.TextLink>
          </Header.Frame>
        </Header>
      </Router>
    );
    expect(getByText("Hello I am a link!")).toBeTruthy();
    expect(getByTestId("header-bg")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("Renders the basic <Header/> without a background", () => {
    const history = createMemoryHistory();
    const { container, getByText, queryByTestId } = render(
      <Router location={history.location} navigator={history}>
        <Header bg={false}>
          <Header.Frame>
            <Header.Logo to="/" src="/logo.svg" alt="Netflix" />
            <Header.ButtonLink to="/">Sign in</Header.ButtonLink>
            <Header.TextLink active="false">Hello I am a link!</Header.TextLink>
          </Header.Frame>
        </Header>
      </Router>
    );
    expect(getByText("Hello I am a link!")).toBeTruthy();
    expect(queryByTestId("header-bg")).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("Renders the full <Header/> with background", () => {
    const history = createMemoryHistory();
    const { container, getByText, getByTestId } = render(
      <Router location={history.location} navigator={history}>
        <Header src="joker1" dontShowOnSmallViewPort>
          <Header.Frame>
            <Header.Group>
              <Header.Logo to="/" src="/images/logo.svg" alt="Netflix" />
              <Header.TextLink active={false} onClick={() => {}}>
                Series
              </Header.TextLink>
              <Header.TextLink active onClick={() => {}}>
                Films
              </Header.TextLink>
            </Header.Group>
            <Header.Group>
              <Header.Search searchTerm="Joker" setSearchTerm={() => {}} />
              <Header.Profile>
                <Header.Picture src="/images/ting.png" />
                <Header.Dropdown>
                  <Header.Group>
                    <Header.Picture src="/images/ting.png" />
                    <Header.TextLink>Ting-Ting Chen</Header.TextLink>
                  </Header.Group>
                  <Header.Group>
                    <Header.TextLink onClick={() => {}}>
                      Sign out
                    </Header.TextLink>
                  </Header.Group>
                </Header.Dropdown>
              </Header.Profile>
            </Header.Group>
          </Header.Frame>

          <Header.Feature>
            <Header.FeatureCallOut>Watch Joker now</Header.FeatureCallOut>
            <Header.Text>Forever alone in a crowd.</Header.Text>
            <Header.PlayButton>Play</Header.PlayButton>
          </Header.Feature>
        </Header>
      </Router>
    );
    expect(getByTestId("search-input")).toBeTruthy();
    expect(getByTestId("search-input").value).toBe("Joker");
    fireEvent.change(getByTestId("search-input"), {
      target: { value: "Simpsons" },
    });
    fireEvent.click(getByTestId("search-click"));

    expect(getByText("Series")).toBeTruthy();
    expect(getByText("Films")).toBeTruthy();
    expect(getByText("Ting-Ting Chen")).toBeTruthy();
    expect(getByText("Watch Joker now")).toBeTruthy();
    expect(getByText("Sign out")).toBeTruthy();
    expect(getByText("Forever alone in a crowd.")).toBeTruthy();
    expect(getByText("Play")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
