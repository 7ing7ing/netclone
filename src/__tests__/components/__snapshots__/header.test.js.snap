// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> Renders the basic <Header/> with a background 1`] = `
<div
  class="header__Background-sc-djltcn-0 dTWthH"
  data-testid="header-bg"
>
  <div
    class="header__Container-sc-djltcn-2 gPcfnc"
  >
    <a
      href="/"
    >
      <img
        alt="Netclone"
        class="header__Logo-sc-djltcn-11 kPTRyU"
      />
    </a>
    <p
      class="header__Link-sc-djltcn-4 hXLfgd"
    >
      Hello I am a link!
    </p>
  </div>
</div>
`;

exports[`<Header/> Renders the basic <Header/> without a background 1`] = `
<div
  class="header__Container-sc-djltcn-2 gPcfnc"
>
  <a
    href="/"
  >
    <img
      alt="Netclone"
      class="header__Logo-sc-djltcn-11 kPTRyU"
      src="/netclone-logo.png"
    />
  </a>
  <a
    class="header__ButtonLink-sc-djltcn-12 dNRoLn"
    href="/"
  >
    Sign in
  </a>
  <p
    class="header__Link-sc-djltcn-4 jRBIhJ"
  >
    Hello I am a link!
  </p>
</div>
`;

exports[`<Header/> Renders the full <Header/> with background 1`] = `
<div
  class="header__Background-sc-djltcn-0 dzbBXF"
  data-testid="header-bg"
  src="joker1"
>
  <div
    class="header__Container-sc-djltcn-2 gPcfnc"
  >
    <div
      class="header__Group-sc-djltcn-1 fqEykS"
    >
      <a
        href="/"
      >
        <img
          alt="Netclone"
          class="header__Logo-sc-djltcn-11 kPTRyU"
          src="/images/netclone-logo.png"
        />
      </a>
      <p
        class="header__Link-sc-djltcn-4 jRBIhJ"
      >
        Series
      </p>
      <p
        class="header__Link-sc-djltcn-4 jRBIhJ"
      >
        Films
      </p>
    </div>
    <div
      class="header__Group-sc-djltcn-1 fqEykS"
    >
      <div
        class="header__Search-sc-djltcn-6 lbveUT"
      >
        <button
          class="header__SearchIcon-sc-djltcn-7 dVIos"
          data-testid="search-click"
        >
          <img
            alt="Search"
            src="/images/icons/search.png"
          />
        </button>
        <input
          class="header__SearchInput-sc-djltcn-9 eGONT"
          data-testid="search-input"
          placeholder="Search films and series"
          value="Joker"
        />
      </div>
      <div
        class="header__Profile-sc-djltcn-10 bVBtkl"
      >
        <button
          class="header__Picture-sc-djltcn-3 wvCYX"
          src="/images/users//images/ting.png.png"
        />
        <div
          class="header__Dropdown-sc-djltcn-5 kjPuDj"
        >
          <div
            class="header__Group-sc-djltcn-1 fqEykS"
          >
            <button
              class="header__Picture-sc-djltcn-3 wvCYX"
              src="/images/users//images/ting.png.png"
            />
            <p
              class="header__Link-sc-djltcn-4 jRBIhJ"
            >
              Ting-Ting Chen
            </p>
          </div>
          <div
            class="header__Group-sc-djltcn-1 fqEykS"
          >
            <p
              class="header__Link-sc-djltcn-4 jRBIhJ"
            >
              Sign out
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="header__Container-sc-djltcn-2 header__Feature-sc-djltcn-13 gPcfnc gGuoEb"
  >
    <p
      class="header__FeatureCallOut-sc-djltcn-15 fCEZRT"
    >
      Watch Joker now
    </p>
    <p
      class="header__Text-sc-djltcn-14 igDnLb"
    >
      Forever alone in a crowd.
    </p>
    <button
      class="header__PlayButton-sc-djltcn-8 kSbuqB"
    >
      Play
    </button>
  </div>
</div>
`;
